imap_processing/hi/hi_l2.py

```diff
@@ -243,19 +243,17 @@ def calculate_ena_intensity(
         np.ones((map_ds["esa_energy_step"].size, map_ds["calibration_prod"].size)),
         coords=[map_ds["esa_energy_step"], map_ds["calibration_prod"]],
     )
-    # TODO: Implement esa energies lookup
-    if esa_energies_path:
-        raise NotImplementedError
-    esa_energy = xr.ones_like(map_ds["esa_energy_step"])
+
+    esa_energy = esa_energy_lookup(esa_energies_path, map_ds["esa_energy_step"].data)
 
     # Convert ENA Signal Rate to Flux
     intensity_vars = {
-        "ena_intensity": map_ds["ena_signal_rates"] / (geometric_factor * esa_energy),
+        "ena_intensity": map_ds["ena_signal_rates"]
+        / (geometric_factor * esa_energy[:, np.newaxis]),
```

Comment:
Break the denominator out into a separate variable so you can re-use it in all of these functions and not recompute it every time?